WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2610,6 +2624,37 @@ public String onAuthHeaderRequested(String url) {
 
     @Override
     public void onEditorFragmentInitialized() {
+        boolean shouldFinishInit = true;
+        // now that we have the Post object initialized,
+        // check whether we have media items to insert from the WRITE POST with media functionality
+        if (getIntent().hasExtra(EXTRA_INSERT_MEDIA)) {
+            List<MediaModel> mediaList = (List<MediaModel>) getIntent().getSerializableExtra(EXTRA_INSERT_MEDIA);
+            if (mediaList != null && !mediaList.isEmpty()) {
```

Comment:
At this point we need to remove `EXTRA_INSERT_MEDIA` from the intent, otherwise this will run again if the activity is re-created.

Reply:
Mindblown by this catch. Thanks!  addressed in 277b69a